File: src/irrgarten/CombatElement.java
package irrgarten;

public class CombatElement {
    private float effect;
    private Integer uses;

    CombatElement(float effect ,Integer uses){
        this.effect = effect;
        this.uses=uses;
    }
    protected float produceEffect(){
        return 0.1f;
    }
    public boolean discard(){

    }

    @Override
    public String toString(){

    }

}

File: JUEGO-RUBY/irrgarten/CombatElement.rb
require_relative 'Dice'
module Irrgarten
    class CombatElement
        def initialize(effect,uses)
            @effect = effect
            @uses = uses
        end
        def produce_effect
            if @uses > 0 
                @uses = @uses -1
                return @effect
            else
                return 0
            end
        end
        def discard()
            return Dice.discard_element(@uses)
        end
        def to_s()
            return "[#@effect,#@uses]"
        end
    end
end
